BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m

```diff
@@ -65,74 +65,105 @@ - (BOXAPIOperation *)createOperation
     if (fieldString.length > 0) {
         queryParameters = @{BOXAPIParameterKeyFields:fieldString};
     }
-    
-    BOXAPIJSONOperation *JSONOperation = [self JSONOperationWithURL:URL
-                                                         HTTPMethod:BOXAPIHTTPMethodGET
-                                              queryStringParameters:queryParameters
-                                                     bodyDictionary:nil
-                                                   JSONSuccessBlock:nil
-                                                       failureBlock:nil];
+
+    BOXAPIOperation *fileOperation = nil;
+    if (self.associateId != nil) {
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                             HTTPMethod:BOXAPIHTTPMethodGET
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:nil
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateID];
+        NSString *requestDirectory = self.requestDirectoryPath;
+        if (requestDirectory.length == 0) {
+            // Emergency fallback for robustness. The request directory path is expected to be set at request configuration time.
+            requestDirectory = NSTemporaryDirectory();
+        }
+        //FIXME: I believe this should name the operation's destination file more uniquely (for example by
+        // pairing the associate ID with the type of operation and file ID), otherwise you can potentially
+        // get thrashing on the same file for various background actions.
```

Comment:
Need your opinion on this FIXME. Would like to take care of it before pushing this, and I do think we should make the path more unique. What do you think?